src/openms/source/FORMAT/XMLFile.cpp

```diff
@@ -152,21 +154,38 @@ namespace OpenMS
       try
       {
         parser->parse(*source);
-        delete(parser);
+        delete parser;
         delete source;
       }
       catch (const xercesc::XMLException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", 
+            String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const xercesc::SAXException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "",
+            String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const XMLHandler::EndParsingSoftly & /*toCatch*/)
       {
-        //nothing to do here, as this exception is used to softly abort the parsing for whatever reason.
+        // nothing to do here except free memory, as this exception is used to
+        // softly abort the parsing for whatever reason.
+        delete parser;
+        delete source;
       }
+      catch (...)
+      {
+        // clean up memory first, then re-throw
+        delete parser;
+        delete source;
+        throw;
```

Comment:
this very likely aborts the program anyways but we should still be nice and clean up memory...